pyqtgraph/functions.py

```diff
@@ -1786,7 +1786,12 @@ def downsample(data, n, axis=0, xvals='subsample'):
     sl[axis] = slice(0, nPts*n)
     d1 = data[tuple(sl)]
     d1.shape = tuple(s)
-    d2 = d1.mean(axis+1)
+    if nanPolicy == 'ignore':
+        d2 = np.nanmean(d1, axis+1)
```

Comment:
Since the simple mean of 'propagate' is the fast solution, I'd propose to make that the default, and to also test for it first.